transformation/src/main/scala/org/broadinstitute/monster/clinvar/ArchiveBranches.scala

```diff
@@ -152,4 +136,38 @@ object ArchiveBranches extends PipelineCoders {
       traitMappings = sideCtx(traitMappingOut)
     )
   }
+
+  def dedupByDate[T: Coder](
+    in: SCollection[(T, Option[LocalDate])],
+    description: String
+  )(getId: T => String): SCollection[T] =
+    in.transform(s"Deduplicate $description elements") {
+      _.groupBy { case (item, _) => getId(item) }.values.flatMap { itemGroup =>
+        val datedItems = itemGroup.flatMap {
+          case (item, date) => date.map(item -> _)
+        }
+        if (datedItems.isEmpty) {
+          // No way to tell what the "right" item is.
+          itemGroup.headOption.map(_._1).toIterable
+        } else {
+          Iterable(datedItems.maxBy(_._2)._1)
```

Comment:
Does this mean we are choosing the most recent items?

Reply:
Yup!